### PR TITLE
Kubaw/ create products services

### PR DESCRIPTION
**Pull request by Wierzb**

### CODE REVIEW

**Review by zuzwoj — APPROVED** [review]
Ok byczku

File: Src/InGreed.Api/Controllers/ProductController.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    [HttpPost]
+    public ActionResult<Product> Create([FromBody] Product product)
+    {
+        if (product is null) return BadRequest();
+        service.CreateProduct(product);
+        return Ok(product);
+    }
+    [HttpPut]
+    public ActionResult<Product> Modify(int productToModifyId, [FromBody] Product product)
```
Comment (portalion) [inline Src/InGreed.Api/Controllers/ProductController.cs]: Add Contract for request

File: Src/InGreed.Api/Controllers/ProductController.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+{
+    private readonly IProductService service;
+
+    public ProductController(IProductService service)
+    {
+        this.service = service;
+    }
+
+    [HttpPost]
+    public ActionResult<Product> Create([FromBody] Product product)
```
Comment (portalion) [inline Src/InGreed.Api/Controllers/ProductController.cs]: Add Contract for request

File: Src/InGreed.Api/Controllers/ProductController.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+    [HttpPut]
+    public ActionResult<Product> Modify(int productToModifyId, [FromBody] Product product)
+    {
+        if (product is null) return BadRequest();
+        service.ModifyProduct(productToModifyId, product);
+        return Ok(product);
+    }
+    [HttpGet]
+    public ActionResult<List<Product>> GetAllProducts()
```
Comment (portalion) [inline Src/InGreed.Api/Controllers/ProductController.cs:34]: Add Contract for request

File: Src/InGreed.Api/Controllers/ProductController.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        List<Product> products = new List<Product>();
+        try { products = service.GetAllProducts().ToList(); }
+        catch (ArgumentException e) { return NotFound(e.Message); }
+        return Ok(products);
+    }
+
+    [HttpGet("{id}")]
+    public ActionResult<Product> GetById(int id)
```
Comment (portalion) [inline Src/InGreed.Api/Controllers/ProductController.cs:43]: Add Contract for request

**Review by portalion — CHANGES_REQUESTED** [review]

**Review by portalion — APPROVED** [review]

**Pull request merged**